Documentation/ApiOverview/Database/QueryBuilder/Index.rst

```diff
@@ -1,1133 +1,1234 @@
+    triggers the magic of the :ref:`RestrictionBuilder
+    <database-restriction-builder>` that adds default restrictions such as
+    :sql:`deleted`, :sql:`hidden`, :sql:`starttime` and :sql:`endtime` when
+    defined in TCA.
 
-*  Similar to :php:`->select()` query types, :php:`->executeQuery()` with
-   :php:`->count()` returns a result object of type :php:`\Doctrine\DBAL\Result`.
-   To fetch the number of rows directly, use :php:`->fetchOne()`.
+*   Similar to :php:`->select()` query types, :php:`->executeQuery()` with
+    :php:`->count()` returns a result object of type :php:`\Doctrine\DBAL\Result`.
+    To fetch the number of rows directly, use :php:`->fetchOne()`.
 
-*  First argument to :php:`->count()` is required, typically :php:`->count(*)`
-   or :php:`->count('uid')` is used, the field
-   name is automatically quoted.
+*   The first argument to :php:`->count()` is required, typically
+    :php:`->count(*)` or :php:`->count('uid')` is used, the field name is
+    automatically quoted.
 
-*  There is no support for `DISTINCT`, a :php:`->groupBy()` has to be used
-   instead.
+*   There is no support for :sql:`DISTINCT`, instead a :php:`->groupBy()` has to
+    be used.
 
-*  If combining :php:`->count()` with a :php:`->groupBy()`, the result may
-   return multiple rows. The order of
-   those rows depends on the used `DBMS`. To ensure same order of result rows
-   on multiple different databases,
-   a :php:`->groupBy()` should thus always be combined with a :php:`->orderBy()`.
+*   If :php:`->count()` is combined with :php:`->groupBy()`, the result may
+    return multiple rows. The order of those rows depends on the used
+    :abbr:`DBMS (Database management system)`. Therefore, to ensure the same
+    order of result rows on multiple different databases, a :php:`->groupBy()`
+    should always be combined with an :php:`->orderBy()`.
 
 
 delete()
 ========
 
-Create a `DELETE FROM` query. The method requires the table name to drop
-data from. Classic usage:
+Create a :sql:`DELETE FROM` query. The method requires the table name from which
+data is to be deleted. Classic usage:
 
-.. code-block:: php
-   :caption: EXT:some_extension/Classes/SomeClass.php
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/Domain/Repository/MyRepository.php
 
-   // use TYPO3\CMS\Core\Utility\GeneralUtility;
-   // use TYPO3\CMS\Core\Database\ConnectionPool;
-   // DELETE FROM `tt_content` WHERE `bodytext` = 'klaus'
-   $queryBuilder = GeneralUtility::makeInstance(ConnectionPool::class)->getQueryBuilderForTable('tt_content');
-   $affectedRows = $queryBuilder
-      ->delete('tt_content')
-      ->where(
-         $queryBuilder->expr()->eq('bodytext', $queryBuilder->createNamedParameter('klaus'))
-      )
-      ->executeStatement();
+    // DELETE FROM `tt_content` WHERE `bodytext` = 'klaus'
+    $queryBuilder = $this->connectionPool->getQueryBuilderForTable('tt_content');
```

Comment:
Please include injection of the ConnectionPool. newbies dont't always have that context

Reply:
yeah I can see how we would need them in many places. However if we just assume `$this->connectionPool` we don't even tell the type

Reply:
Yes, using full classes would blow up examples. First I haven't `$this->connectionPool->getQueryBuilderForTable()` in the snippets, but is relevant as the query builder instance depends on the table name to get the correct connection.

In the query builder chapter at least the instantiation section should be read. There are so many cross-references between the sections, so you should read at least the introduction to get a clue what's going on.

Anyway, we can add a comment in each example (just before `$this->connectionPool->...` to look up the instatiation section on top of the chapter.

Reply:
I would prefere to have a comment linking how to get the connection Pool, this is a topic where many newbies might get frustrated on.

Reply:
Yup, but a newbie should read the chapter and not just copy/paste. As I said, many cross-references between the different sections.

But I will add a comment in the snippets.

Reply:
I would add them outside the snippets and link them properly?

We have the problem that ppl are searching for something and then they will land on one of these pages. We cannot assume they start with A.

Reply:
Okay, then above all snippets links... That'll be fun ...

query builder is special. ppl should know what they are doing.

Reply:
Done it below, above does not fit.